README.markdown

```diff
@@ -449,6 +449,22 @@ If you want to stop processing jobs, but want to leave the worker running
 (for example, to temporarily alleviate load), use `USR2` to stop processing,
 then `CONT` to start it again.
 
+#### Signals on Heroku
+
+When shutting down processes, Heroku sends every process a TERM signal at the
+same time. By default this causes an immediate shutdown of any running job
+leading to frequent `Resque::TermException` errors.  For short running jobs, a simple
+solution is to give a small amount of time for the job to finish
+before killing it.
+
+To accomplish this set the following environment variables:
+
+* `RESQUE_PRE_SHUTDOWN_TIMEOUT` - The time between the parent receiving a shutdown signal (TERM by default) and it sending that signal on to the child process. Designed to give the child process
+time to complete before being forced to die.
+
+* `RESQUE_SHUTDOWN_SIGNAL` - The signal sent to the child process
```

Comment:
We've been handing `Resque::DirtyExit` by requeuing jobs just as we do with `Resque::TermException`, but this could be caused by not handling this signal correctly. We'll try and see tomorrow if we fix it propertly so we don't need to catch DirtyExit.